src/EmbedIO/Security/IPBanningModule.cs

```diff
@@ -0,0 +1,301 @@
+﻿using EmbedIO.Utilities;
+using Swan;
+using Swan.Logging;
+using Swan.Threading;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Security
+{
+    /// <summary>
+    /// A module for ban IPs that show the malicious signs, based on scanning log messages.
+    /// </summary>
+    /// <seealso cref="WebModuleBase" />
+    public class IPBanningModule : WebModuleBase, ILogger
```

Comment:
The fact that this class implements `ILogger` is an implementation detail. As such, it should not be part of its public surface.

The `ILogger` implementation would be better moved to an embedded, private class, of which an instance is created in the costructor.

Reply:
I already create an inner private class.